lib/azure/core/http/http_request.rb

```diff
@@ -92,19 +96,20 @@ def initialize(method, uri, options_or_body = {})
         def with_filter(filter=nil, &block)
           filter = filter || block
           if filter
-            old_impl = self._method(:call)
+            @has_retry_filter = filter.is_a? Azure::Core::Http::RetryPolicy
+            original_call = self._method(:call)
 
             # support 1.8.7 (define_singleton_method doesn't exist until 1.9.1)
```

Comment:
Though we don't officially announce we support Ruby 2.0 and below, this code can keep us safe when people call with lower version and brings backward compatibility. It is not in the scope of this change, if you still want to remove it, I can do it in another PR of dropping the support for Ruby 2.0 and lower.